lib/Backends/NNPI/NNPIDeviceManager.cpp

```diff
@@ -130,20 +130,24 @@ Error NNPIDeviceManager::init() {
     if (deviceOptions_.enabledDeviceTracing) {
       deviceTracing_ = NNPIDeviceTracing::getForDevice(deviceId_);
     }
+    NNPIDeviceInfo deviceInfo;
+    LOG_NNPI_INF_ERROR_RETURN_LLVMERROR(
+        nnpiDeviceGetInfo(deviceId_, &deviceInfo),
+        "Failed to get NNPI Device Info");
+    maxMemoryBytes_ =
+        static_cast<uint64_t>(deviceInfo.totalUnprotectedMemory) * KB;
```

Comment:
Is `totalUnprotectedMemory` the size of DRAM? If so, perhaps we should set `maxMemoryBytes_` to what's actually usable?

Reply:
That's the usable portion of the DRAM for deployed networks without going into special directives for protected memory